### PR TITLE
added solution

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/main/java/core/basesyntax/BooleanExpression.java [hunk 1]
```diff
@@ -14,6 +14,7 @@
      * без использования конструкций if-else</p>
      */
     public boolean booleanExpression(boolean a, boolean b, boolean c, boolean d) {
-        return false;
+        return (a == b & a != c & a != d) || (a == c & a != b & a != d)
+                || (a == d & a != c & a != b);
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/BooleanExpression.java]: try to simplify it with binary operators

Comment (ghost) [reply src/main/java/core/basesyntax/BooleanExpression.java]: done

File: src/main/java/core/basesyntax/BooleanExpression.java [hunk 2]
```diff
@@ -14,6 +14,7 @@
      * без использования конструкций if-else</p>
      */
     public boolean booleanExpression(boolean a, boolean b, boolean c, boolean d) {
-        return false;
+        return ((a ^ b) && (c ^ d)) || ((a ^ c) && (b ^ d));
+
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/BooleanExpression.java]: remove empty line

Comment (ghost) [reply src/main/java/core/basesyntax/BooleanExpression.java]: done

File: src/main/java/core/basesyntax/BooleanExpression.java [hunk 3]
```diff
@@ -14,6 +14,6 @@
      * без использования конструкций if-else</p>
      */
     public boolean booleanExpression(boolean a, boolean b, boolean c, boolean d) {
-        return false;
+        return ((a ^ b) && (c ^ d)) || ((a ^ c) && (b ^ d));
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/BooleanExpression.java]: remove redundant brackets

Comment (ghost) [reply src/main/java/core/basesyntax/BooleanExpression.java]: done

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request closed without merging**